### DISCUSSION TITLE
How to use my own Assistant?

### DISCUSSION THREAD

**Discussion by Dekaltrim** [betalgo/openai#465] [Q&A]
Hi!

I have get it working with text and image. But i can´t figure out how to use my own assistant i have created in openai.com?

Is it possible to fine-tune in openai api and use assistant with this library?

Best regards,
Klas

**Comment by dmki** [comment]
As far as I understand, you can't use the assistant you have created in openai.com. You can re-create assistant using API, and then use it by API. And because assistants created manually on a website only have URL and not internal ID, you can't list / consume them, at least yet.

The main part, though, is that no SDK supports assistants yet, even though it's perhaps the most important update after introducing GPT4 (for me, anyway). But what looks simple in the web, adds many concepts to SDK - assistants, threads, runs, files... So will take time to review and test. Plus it was Christmas, and many people didn't touch their computers for weeks.

So, I hope that we'll see the support for assistants in around 2 weeks, as the code is already there in pull request and is waiting to be reviewed / tested.

Answering your question about how to use your assistant - your code will have to create / retrieve existing assistant, then thread, then run, and then operate it. I'm sure there will be a code snippet in docs once it will be implemented.

**Comment by Artein** [comment]
@Dekaltrim you can try the new Assistants API in 8.2.0-beta
